### PR TITLE
Add typescript

### PR DESCRIPTION
**Pull request by bbovenzi**
Add typescript support for the current UI project in `/airflow/www`.

- Include type checking as part of the `yarn lint` checks
- Add example `.tsx` and `.test.tsx` files with the `Time` component and fix some associated typing issues

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragement file, named `{pr_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Review by pierrejeambrun — APPROVED** [review]
Really nice!

I guess that `ts` is the new standard from now on ? (Also updating existing code each time we have the opportunity to)

**Comment by bbovenzi** [comment]
> Really nice!
> 
> I guess that `ts` is the new standard from now on ? (Also updating existing code each time we have the opportunity to)

Yep. That's the plan. We can migrate js files over time and across PRs.

File: airflow/www/.eslintrc [hunk 1]
```diff
@@ -5,6 +5,32 @@
   "rules": {
```
Comment (mik-laj) [inline airflow/www/.eslintrc:5]: Should we also add [typescript-eslint](https://github.com/typescript-eslint/typescript-eslint)?

**Review by dstandish — APPROVED** [review]

File: airflow/www/.eslintrc (hunk 1 above)
Comment (bbovenzi) [reply airflow/www/.eslintrc:5]: Yeah, I was playing around with it. Let me make sure it's in this PR.

Comment (bbovenzi) [reply airflow/www/.eslintrc:5]: Added!

**Review by mik-laj — APPROVED** [review]

**Comment by uranusjr** [comment]
🎉

**Comment by potiuk** [comment]
Cool! Maybe finally some chance to contribute to the UI part :)

**Pull request merged**